ofrak_core/ofrak/core/patch_maker/modifiers.py

```diff
@@ -209,15 +209,19 @@ async def modify(self, resource: Resource, config: SegmentInjectorModifierConfig
 
             injection_tasks.append((region.resource, BinaryInjectorModifierConfig(patches)))
 
-        all_decendants = await resource.get_descendants()
         for injected_resource, injection_config in injection_tasks:
             result = await injected_resource.run(BinaryInjectorModifier, injection_config)
-            parents_to_delete = list(
-                filter(lambda r: r.get_id() in result.resources_modified, all_decendants)
+            allowed_to_delete = result.resources_modified.difference(
```

Comment:
I find `allowed_to_delete` a confusing variable name, but I'm struggling to come up with a better idea to express what is happening here. 

Perhaps a descriptive comment could help?